### PR TITLE
[DependencyInjection] Only dump array keys when value is not a list

### PR DESCRIPTION
**Pull request by ruudk**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  |no 
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

When the value is a list, there is no point in dumping the keys.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1793,12 +1793,13 @@ private function getDefinitionsFromArguments(array $arguments, \SplObjectStorage
     private function dumpValue(mixed $value, bool $interpolate = true): string
     {
         if (\is_array($value)) {
+            $isList = array_is_list($value);
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: I would move it after the `if` that does not need it.

Comment (ruudk) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Oops, that makes much sense. Fixed.

**Comment by nicolas-grekas** [comment]
While you're at PhpDumper, can you please send a PR to make all closures (function+fn) non-static? In 6.3, this is visual debt because they're declared in static methods only.

**Comment by nicolas-grekas** [comment]
Thank you @ruudk.

**Pull request merged**